### ISSUE TITLE
API Gateway doesn't appear to work

### ISSUE THREAD

**Issue by mattmcla** [localstack/localstack#692]
I've been attempting to get API Gateway to work with Lambda but have been continuously running into issues. I have successfully set up the lambda function and am able to invoke it via aws cli.

http://localhost:4567/restapis returns
`{"item": [{"description": "Primary Gateway", "createdDate": 1522357512, "id": "89A-Z7482723", "name": "main-gateway"}]}`
 
`curl -v http://localhost:4567/restapis/89A-Z7482723/test/_user_request_/` returns
```
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 4567 (#0)
> GET /restapis/89A-Z7482723/test/_user_request_/ HTTP/1.1
> Host: localhost:4567
> User-Agent: curl/7.56.1
> Accept: */*
>
< HTTP/1.1 404 Not Found
< Server: BaseHTTP/0.3 Python/2.7.13
< Date: Thu, 29 Mar 2018 21:10:58 GMT
< Content-Length: 94
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH
< Access-Control-Allow-Headers: authorization,content-type,content-md5,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent
<
* Connection #0 to host localhost left intact
{"message": "Unable to find integration for path /restapis/89A-Z7482723/test/_user_request_/"}%
```
`aws --endpoint-url=http://localhost:4567 apigateway get-resources --rest-api-id 89A-Z7482723` returns:
```
An error occurred (500) when calling the GetResources operation (reached max retries: 4): <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request.  Either the server is overloaded or there is an error in the application.</p>
```
The last call using the aws cli generates the following logs:

```
Error on request:
Traceback (most recent call last):
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/werkzeug/serving.py", line 270, in run_wsgi
execute(self.server.app)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/werkzeug/serving.py", line 258, in execute
application_iter = app(environ, start_response)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/moto/server.py", line 93, in __call__
return backend_app(environ, start_response)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/flask/app.py", line 1836, in __call__
return self.wsgi_app(environ, start_response)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/flask/app.py", line 1820, in wsgi_app
response = self.make_response(self.handle_exception(e))
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/flask/app.py", line 1403, in handle_exception
reraise(exc_type, exc_value, tb)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
response = self.full_dispatch_request()
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
reraise(exc_type, exc_value, tb)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
rv = self.dispatch_request()
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/moto/core/utils.py", line 139, in __call__
result = self.callback(request, request.url, {})
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/moto/apigateway/responses.py", line 57, in resources
resources = self.backend.list_resources(function_id)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/moto/apigateway/models.py", line 468, in list_resources
api = self.get_rest_api(function_id)
File "/opt/code/localstack/.venv/lib/python2.7/site-packages/moto/apigateway/models.py", line 457, in get_rest_api
rest_api = self.apis[function_id]
KeyError: u'89A-Z7482723'
```

Command used to start localstack: `LAMBDA_REMOTE_DOCKER=false LAMBDA_EXECUTOR=docker-reuse DEFAULT_REGION=us-east-1 SERVICES=s3,apigateway,lambda DEBUG=1 localstack start --docker`

localstack version 0.8.5

**Comment by mattmcla** [comment]
I just updated my copy of `localstack` and re-ran my build script. It's still failing when I try to curl the endpoint with an error of `{"message": "Unable to find integration for path /restapis/8627A-Z0A-Z413/test/_user_request_/"}` but I noticed it query itself to get a list of integrations `/restapis/8627A-Z0A-Z413/resources/812303A-Z983/methods/GET/integration` However, my gateway method is ANY, which should match any verb. I curled `/restapis/8627A-Z0A-Z413/resources/812303A-Z983/methods/ANY/integration` manually and it gave me back the integration. When I curled the same thing but with GET I get back `null`
